cmd/README.md

```diff
@@ -0,0 +1,738 @@
+
+Set up gas costs. In `precompile/params.go` we have `writeGasCostPerSlot` and `readGasCostPerSlot`. This is a good starting point for estimating gas costs. 
+
+``` go
+// Gas costs for stateful precompiles
+const (
+	writeGasCostPerSlot = 20_000
+	readGasCostPerSlot  = 5_000
+)
+```
+
+For example, we will be setting and getting our greeting in one slot so we can define the gas costs as follows. 
+
+``` go
+	SayHelloGasCost uint64    = 20_000 
+	SetGreetingGasGost uint64 = 5000
+```
+
+
+**Example:** 
+The sha256 precompile computes gas with the following equation
+``` go
+// This method does not require any overflow checking as the input size gas costs
+// required for anything significant is so high it's impossible to pay for.
+func (c *sha256hash) RequiredGas(input []byte) uint64 {
+	return uint64(len(input)+31)/32*params.Sha256PerWordGas + params.Sha256BaseGas
+}
+```
+
+## Step 3: Add Custom Code
+
+Ok time to `CTRL F` throughout the file with `CUSTOM CODE STARTS HERE` to find the areas in the precompile that we need to modify. 
+
+![](2022-09-01-22-48-26.png)
+
+If we use any of the following imports we can remove the reference imports.
+We can come back if needed to remove. Next we see this in `Equals()`.
+
+![](2022-09-01-22-51-30.png)
+
+We can skip this step since our HelloWorldConfig struct looks like this.
+
+```
+// HelloWorldConfig implements the StatefulPrecompileConfig
+// interface while adding in the HelloWorld specific precompile address.
+type HelloWorldConfig struct {
+	UpgradeableConfig
+}
+```
+
+**Optional Note** 
+
+If our HelloWorldConfig wrapped another config in its struct to implement the StatefulPrecompileConfig
+like so, 
+
+``` go
+// HelloWorldConfig implements the StatefulPrecompileConfig
+// interface while adding in the IHelloWorld specific precompile address.
+
+type HelloWorldConfig struct {
+	UpgradeableConfig
+    AllowListConfig
+}
+```
+
+We would have to modify the `Equal()` function as follows: 
+
+![](2022-09-01-22-54-22.png)
+
+The next place we see the `CUSTOM CODE STARTS HERE` is in `Configure()`
+Let's set it up. 
+
+``` go
+// Configure configures [state] with the initial configuration.
+func (c *HelloWorldConfig) Configure(_ ChainConfig, state StateDB, _ BlockContext) {
+	// CUSTOM CODE STARTS HERE
+	// This will be called in the first block where HelloWorld stateful precompile is enabled.
+	// 1) If BlockTimestamp is nil, this will not be called
+	// 2) If BlockTimestamp is 0, this will be called while setting up the genesis block
+	// 3) If BlockTimestamp is 1000, this will be called while processing the first block
+	// whose timestamp is >= 1000
+	//
+	// Set the initial value under [common.BytesToHash([]byte("storageKey")] to "Hello World!"
+	res := common.LeftPadBytes([]byte("Hello World!"), common.HashLength)
+	state.SetState(HelloWorldAddress, common.BytesToHash([]byte("storageKey")), common.BytesToHash(res))
+}
+```
+We also see a `Verify()` function. We can leave this as is for now. 
```

Comment:
Can we provide an explanation of how `Verify()` works here (there's an explanation in `./precompile/stateful_precompile_config.go` and explain that we leave this as a no-op because there is no invalid configuration for the `HelloWorldConfig`

Reply:
Can we explain why there is no verification needed within this function and then ask the user to delete the CUSTOM CODE comment and provide a copy pastable version of this code with the comment removed that the user can replace the code with?

Reply:
Yes thanks